Assets/SequenceSDK/WaaS/Tests/WaaSSessionManagementTests.cs

```diff
@@ -125,5 +125,57 @@ public async Task TestGetSessionAuthProof()
                 tcs.TrySetException(new Exception(error));
             });
         }
+
+        [Test]
+        public async Task GetIdTokenTest()
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            var login = SequenceLogin.GetInstance();
+
+            login.OnLoginFailed += (error, method, email, methods) =>
+            {
+
+                Assert.Fail("Login failed: " + error);
+                tcs.TrySetResult(false);
```

Comment:
TrySetException is more effective for error diagnosis. If this test (in its current state) were to fail, it would fail with the message:

> Expected: True
> But was:  False

which doesn't help us identify where it failed; we'd need to use a debugger.